src/hooks/http.ts

```diff
@@ -191,11 +214,14 @@ export class Http {
       if (isRequestToAwsService && !isKeepHeadersOn()) {
         const { awsXAmznTraceId } = getAWSEnvironment();
         const traceId = getPatchedTraceId(awsXAmznTraceId);
-        options.headers['X-Amzn-Trace-Id'] = traceId;
+        headers && (headers['X-Amzn-Trace-Id'] = traceId);
```

Comment:
Was this change needed? there is a default value for headers in line 205

Reply:
Not use options is the important change here :)